### PR TITLE
114 lint non relevant cidr in rule

### PR DESCRIPTION
**Pull request by ShiriMoran**

### CODE REVIEW

File: pkg/linter/lintRuleNonRelevantCIDRCommon.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+			continue // no use in executing lint on dummy vpcs
+		}
+		vpcAdrressRange := config.VPC.AddressRange()
+		filterLayer := config.GetFilterTrafficResourceOfKind(filterLayerName)
+		rules, err := filterLayer.GetRules()
+		if err != nil {
+			return nil, err
+		}
+		var isDisjoint, nonDisjointNonContained bool
+		for _, rule := range rules {
```
Comment (haim-kermany) [inline pkg/linter/lintRuleNonRelevantCIDRCommon.go:39]: comment the new code

**Review by haim-kermany — CHANGES_REQUESTED** [review]

File: pkg/linter/lintRuleNonRelevantCIDRCommon.go (hunk 1 above)
Comment (ShiriMoran) [reply pkg/linter/lintRuleNonRelevantCIDRCommon.go:40]: done

**Review by haim-kermany — APPROVED** [review]

File: pkg/linter/lintRuleNonRelevantCIDRCommon.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	strSuffix = fmt.Sprintf("the VPC's Address Range %s\n\tRule's details: %s",
+		finding.vpcResource.AddressRange().String(), rule.RuleDesc)
+	return strPrefix + issueStr + strSuffix
+}
+
+// for json:
+type rulesNonRelevantCIDRJSON struct {
+	Rule            vpcmodel.RuleOfFilter `json:"vpc_name"`
+	VpcName         string                `json:"rule_details"`
```
Comment (zivnevo) [inline pkg/linter/lintRuleNonRelevantCIDRCommon.go:87]: json tags are wrong

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/linter/lintRuleNonRelevantCIDRSG.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+func (lint *ruleNonRelevantCIDRSGLint) check() error {
+	rulesNonRelevantCIDRFound, err := findRuleNonRelevantCIDR(lint.configs, vpcmodel.SecurityGroupLayer)
+	if err != nil {
+		return err
+	}
+	for _, ruleNonRelevantCIDRFound := range rulesNonRelevantCIDRFound {
+		lint.addFinding(&ruleNonRelevantCIDRFound)
+	}
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/linter/lintRuleNonRelevantCIDRSG.go]: done

File: pkg/linter/lintRuleNonRelevantCIDRCommon.go (hunk 2 above)
Comment (ShiriMoran) [reply pkg/linter/lintRuleNonRelevantCIDRCommon.go:87]: fixed

File: pkg/linter/lintRuleNonRelevantCIDRCommon.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+			continue // no use in executing lint on dummy vpcs
+		}
+		vpcAddressRange := config.VPC.AddressRange()
+		filterLayer := config.GetFilterTrafficResourceOfKind(filterLayerName)
+		rules, err := filterLayer.GetRules()
+		if err != nil {
+			return nil, err
+		}
+		// for ingress dst addresses ips within the VPC, for egress src
+		for _, rule := range rules {
```
Comment (zivnevo) [inline pkg/linter/lintRuleNonRelevantCIDRCommon.go:40]:
```suggestion
		for i := range rules {
			rule := &rules[i]
```
Avoid copying structs

**Review by zivnevo — CHANGES_REQUESTED** [review]

File: pkg/linter/lintRuleNonRelevantCIDRCommon.go (hunk 4 above)
Comment (ShiriMoran) [reply pkg/linter/lintRuleNonRelevantCIDRCommon.go:40]: right. I'll do it in the remaining place as well

Comment (ShiriMoran) [reply pkg/linter/lintRuleNonRelevantCIDRCommon.go:40]: done in all relevant places

File: pkg/linter/lintBlockedTCPReposnse.go [hunk 5]
```diff
@@ -41,11 +41,12 @@ func (lint *blockedTCPResponseLint) lintDescription() string {
...
-	for _, nodesConn := range lint.nodesConn {
-		for _, line := range nodesConn.GroupedConnectivity.GroupedLines {
-			tcpRspDisable := line.CommonProperties.Conn.TCPRspDisable
+	for i := range lint.nodesConn {
+		for i2 := range lint.nodesConn[i].GroupedConnectivity.GroupedLines {
+			tcpRspDisable := lint.nodesConn[i].GroupedConnectivity.GroupedLines[i2].CommonProperties.Conn.TCPRspDisable
 			if !tcpRspDisable.IsEmpty() {
-				lint.addFinding(&blockedTCPResponseConn{src: line.Src, dst: line.Dst, tcpRspDisable: tcpRspDisable})
+				lint.addFinding(&blockedTCPResponseConn{src: lint.nodesConn[i].GroupedConnectivity.GroupedLines[i2].Src,
+					dst: lint.nodesConn[i].GroupedConnectivity.GroupedLines[i2].Dst, tcpRspDisable: tcpRspDisable})
```
Comment (zivnevo) [inline pkg/linter/lintBlockedTCPReposnse.go:49]: No struct copying took place here before. The map values in `nodesConn` were pointers, and the slice value in `GroupedConnectivity.GroupedLines` were also pointers.

Old code was therefore better (same efficiency, better readability).

Please check all changes in the last commit, and see where pointers are being used (no change needed) and where structs are actually being copied.

**Review by zivnevo — CHANGES_REQUESTED** [review]

**Review by zivnevo — APPROVED** [review]

**Pull request merged**